### PR TITLE
Left sig right sig match failure

### PR DESCRIPTION
**Pull request by jhoneycutt**
## Submitter Checklist:
Resolves https://github.com/brave/brave-browser/issues/6545

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

1. Launch with `--enable-logging=stderr --vmodule=*rewards*=6 --rewards=staging=true,reconcile-interval=10,short-retries=true` to see log messages, use staging env, set short contribution interval, and short retry interval.
2. Recover wallet with 30 BAT
3. Set AC amount to Up to 20 BAT
4. Add verified and non verified publishers to AC table
5. Add verified recurring tips in this order (5, 10, 10, 10, 1)
6. Wait for monthly contribution to trigger
7. You should see 26 BAT goes through for monthly and 4 BAT from AC
8. Watch terminal and verify that no "Left sig != right sig" error messages appear.
9. Verify that all BAT was delivered to appropriate recipients.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by jhoneycutt** [comment]
CI shows several unrelated test failures. However, those failures cause the browser tests to abort early, meaning that the relevant browser tests to this PR are unable to run. The network audit step also fails for unrelated reasons.

File: vendor/bat-native-ledger/src/bat/ledger/internal/contribution/phase_two.cc [hunk 1]
```diff
@@ -245,7 +246,38 @@ void PhaseTwo::PrepareBatch(
       callback);
 }
 
+void PhaseTwo::PrepareBallots(
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/contribution/phase_two.cc]: is this intentionally that you named it the same as a function above?

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/contribution/phase_two.cc]: because they are doing different things

Comment (jhoneycutt) [reply vendor/bat-native-ledger/src/bat/ledger/internal/contribution/phase_two.cc]: No, that was unintentional. I'll rename this new function.

**Comment by jhoneycutt** [comment]
Failing CI steps are `npm run audit_deps` which are unrelated to this PR.

**Comment by bsclifton** [comment]
Updated milestone to be 0.74.x 😄 (since that is what master is, when this was merged)

**Pull request merged**